platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/PlatformStateValueAccumulator.java

```diff
@@ -0,0 +1,484 @@
+        this.lastFrozenTime = lastFrozenTime;
+        lastFrozenTimeUpdated = true;
+    }
+
+    /**
+     * Get the first software version where the birth round migration happened, or null if birth round migration has not
+     * yet happened.
+     *
+     * @return the first software version where the birth round migration happened
+     */
+    @Override
+    @Nullable
+    public SoftwareVersion getFirstVersionInBirthRoundMode() {
+        return firstVersionInBirthRoundMode;
+    }
+
+    /**
+     * Set the first software version where the birth round migration happened.
+     *
+     * @param firstVersionInBirthRoundMode the first software version where the birth round migration happened
+     */
+    @Override
+    public void setFirstVersionInBirthRoundMode(final SoftwareVersion firstVersionInBirthRoundMode) {
+        this.firstVersionInBirthRoundMode = firstVersionInBirthRoundMode;
+        firstVersionInBirthRoundModeUpdated = true;
+    }
+
+    /**
+     * Get the last round before the birth round mode was enabled, or -1 if birth round mode has not yet been enabled.
+     *
+     * @return the last round before the birth round mode was enabled
+     */
+    @Override
+    public long getLastRoundBeforeBirthRoundMode() {
+        return lastRoundBeforeBirthRoundMode;
+    }
+
+    /**
+     * Set the last round before the birth round mode was enabled.
+     *
+     * @param lastRoundBeforeBirthRoundMode the last round before the birth round mode was enabled
+     */
+    @Override
+    public void setLastRoundBeforeBirthRoundMode(final long lastRoundBeforeBirthRoundMode) {
+        this.lastRoundBeforeBirthRoundMode = lastRoundBeforeBirthRoundMode;
+        lastRoundBeforeBirthRoundModeUpdated = true;
+    }
+
+    /**
+     * Get the lowest judge generation before the birth round mode was enabled, or -1 if birth round mode has not yet
+     * been enabled.
+     *
+     * @return the lowest judge generation before the birth round mode was enabled
+     */
+    @Override
+    public long getLowestJudgeGenerationBeforeBirthRoundMode() {
+        return lowestJudgeGenerationBeforeBirthRoundMode;
+    }
+
+    /**
+     * Set the lowest judge generation before the birth round mode was enabled.
+     *
+     * @param lowestJudgeGenerationBeforeBirthRoundMode the lowest judge generation before the birth round mode was
+     *                                                  enabled
+     */
+    @Override
+    public void setLowestJudgeGenerationBeforeBirthRoundMode(final long lowestJudgeGenerationBeforeBirthRoundMode) {
+        this.lowestJudgeGenerationBeforeBirthRoundMode = lowestJudgeGenerationBeforeBirthRoundMode;
+        lowestJudgeGenerationBeforeBirthRoundModeUpdated = true;
+    }
+
+    public boolean isAddressBookUpdated() {
+        return addressBookUpdated;
+    }
+
+    public boolean isPreviousAddressBookUpdated() {
+        return previousAddressBookUpdated;
+    }
+
+    public boolean isRoundUpdated() {
```

Comment:
`isRoundUpdated` and `isConsensusTimestampUpdated` are both unused.

What is going on with these values? They are being set and used in different places throughout the code base, but aren't included in the protobuf?

Reply:
That's actually was the reason for a test failure. I fixed that.